text/0000-declarative-nonstandard-event-binding.md

```diff
@@ -0,0 +1,125 @@
+
+```html
+<!-- Event as part of directive -->
+<third-party lwc:external lwc:on:Foo-Bar-BAZ={handleConnected}></third-party>
+
+<!-- Event as part of attribute name (special treatment for lwc:external) -->
+<third-party lwc:external onFoo-Bar-BAZ={handleConnected}></third-party>
+
+<!-- Event and listener as part of attribute value -->
+<third-party lwc:external lwc:on="Foo-Bar-BAZ:handleConnected"></third-party>
+```
+
+This is an example of a less declarative approach but should also be useful for things like creating
+components from metadata:
+
+```js
+// foo.js
+class Foo extends LightningElement {
+    handlers = {
+        'kebab-case': (event) => console.log("'kebab-case' handled"),
+        camelCase: (event) => console.log("'camelCase' handled"),
+        CAPScase: (event) => console.log("'CAPScase' handled"),
+        PascalCase: (event) => console.log("'PascalCase' handled"),
+    };
+}
+```
+
+```html
+<!-- foo.html -->
+<x-bar lwc:external lwc:on={handlers}></x-bar>
+```
+
+## Motivation
+
+The primary motivation for this feature is to enable component authors to listen for non-standard
+event names in a declarative manner without having to resort to calls to `addEventListener()`. This
+should be especially useful when interfacing with non-LWC components that do not follow LWC
+conventions, which should become more common with the recent introduction of native web component
+support through the use of the `lwc:external` directive.
+
+## Detailed design
+
+The usage of this feature will be restricted to custom elements using `lwc:external`.
+
+The design will be based on the agreed upon API shape, which is still under discussion.
+
+## Drawbacks
+
+A drawback of this feature is that component authors of LWC components may be encouraged to
+implement non-standard event names that requires the use of `lwc:on` for declarative consumption.
+This is not desirable because it makes static analysis more difficult. Conversely, consumers might
+use this directive when they could just as easily have used a standard `on*` attribute in their
+template. These scenarios could be avoided, however, by restricting the usage of `lwc:on` to custom
+elements that use the `lwc:external` directive.
+
+## Alternatives
+
+### lwc:spread
+
+An alternative design that was considered was to build this feature into the existing `lwc:spread`
+directive which currently only supports the setting of property values. In fact, `lwc:spread` can be
+used to listen for standard events like `click` or `focus` by assigning event listeners to the
+`onclick` and `onfocus` properties. However, adding support for non-standard event names would add
+complexity to `lwc:spread`. Introducing a new directive also makes it easier to restrict the usage
+of `lwc:on` to `lwc:external` components.
+
+## lwc:on=`${eventName}:${listenerName}`
+
+Encoding both the event name and event handler name as a string using some delimiter would allow us

```

Comment:
I'm not a big fan of `lwc:on`, especially since Svelte v5 is actually moving [away from their existing `on:click` syntax](https://svelte-5-preview.vercel.app/docs/event-handlers) towards a syntax more like ours, since it plays nicer with their equivalent of `lwc:spread`.